### DISCUSSION TITLE
Error when installing from source code(Github)

### DISCUSSION THREAD

**Discussion by M-Amrollahi** [ultralytics/ultralytics#4627] [Q&A]
I was trying to install the ultralytics from source code on colab with
```python
!git clone https://github.com/ultralytics/ultralytics.git
!cd ultralytics
!pip install -e .

import ultralytics
ultralytics.checks()

```
So I got:

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-9-6a4379890183>](https://localhost:8080/#) in <cell line: 2>()
      1 #from ultralytics import YOLO
----> 2 ultralytics.checks()

AttributeError: module 'ultralytics' has no attribute 'checks'
```

Where am I wrong?

**Edit:** It runs ok when I removed the ```-e``` flag from ```pip```.

**Comment by ExtReMLapin** [comment]
https://github.com/ultralytics/ultralytics/issues/3259

**Reply by glenn-jocher** [reply]
@ExtReMLapin hello! It looks like you're encountering an issue with the editable install (`-e` flag) of the Ultralytics YOLOv8 repository. The error you're seeing suggests that the `ultralytics` module doesn't have an attribute named `checks`, which could be due to an incomplete or incorrect installation.

When you clone the repository and install it in editable mode, make sure you're in the correct directory before running the `pip install -e .` command. The `cd ultralytics` command in your snippet won't change the directory in the Colab environment for subsequent commands. Instead, you should use the `%cd` magic command in Colab like this:

```python
!git clone https://github.com/ultralytics/ultralytics.git
%cd ultralytics
!pip install -e .
```

This ensures that the installation is being executed in the correct directory. If you're still having issues, you might want to try a regular install without the `-e` flag, as you've discovered.

For more detailed instructions and troubleshooting, please refer to our documentation at https://docs.ultralytics.com. If the problem persists, feel free to open an issue on the GitHub repository with the details, and we'll be happy to help you out. 😊
